pkg/util/mount/mount_linux_test.go

```diff
@@ -1220,6 +1221,29 @@ func TestBindSubPath(t *testing.T) {
 			},
 			expectError: false,
 		},
+		{
+			name: "subpath-dir-readonly",
+			prepare: func(base string) ([]string, string, string, error) {
+				volpath, _ := getTestPaths(base)
+				subpath := filepath.Join(volpath, "dir0")
+				return nil, volpath, subpath, os.MkdirAll(subpath, defaultPerm)
+			},
+			expectError: false,
+			readOnly:    true,
+		},
+		{
+			name: "subpath-file-readonly",
+			prepare: func(base string) ([]string, string, string, error) {
+				volpath, _ := getTestPaths(base)
```

Comment:
In these test cases the volpath is not readOnly right? Should we add two more tests here that test ReadOnly Subpath Mount for dir/file for when volPath is readOnly?

Reply:
but could it have different behavior whether the permissions on the volume are read only or not? Almost like doing the same test but in a "different environment"

Reply:
added